include/deal.II/grid/tria_accessor.h

```diff
@@ -2489,6 +2496,126 @@ class CellAccessor :  public TriaAccessor<dim,dim,spacedim>
+   * has_periodic_neighbor() for the @c ith face prior to using this function.
+   * The behavior of periodic_neighbor() is similar to neighbor(), in
+   * the sense that the returned cell has at most the same level of refinement
+   * as the current cell. On distributed meshes, by calling
+   * Triangulation::add_periodicity(),
+   * we can make sure that the element on the other side of the periodic
+   * boundary exists in this rank as a ghost cell or a locally owned cell.
+   */
+  TriaIterator<CellAccessor<dim, spacedim> >
+  periodic_neighbor (const unsigned int i) const;
+
+  /**
+   * For a cell whose @c ith face is not at a boundary, this function returns
+   * the same result as neighbor(). If the @c ith face is at a periodic boundary
+   * this function returns the same result as periodic_neighbor(). If neither of
+   * the aforementioned conditions are met, i.e. the @c ith face is on a
+   * nonperiodic boundary, an exception will be thrown.
+   */
+  TriaIterator<CellAccessor<dim, spacedim> >
+  neighbor_or_periodic_neighbor (const unsigned int i) const;
+
+  /**
+   * Returns an iterator to the periodic neighbor of the cell at a given
+   * face and subface number. The general guidelines for using this function
+   * is similar to the function neighbor_child_on_subface(). The
+   * implementation of this function is consistent with
+   * periodic_neighbor_of_coarser_periodic_neighbor(). For instance,
+   * assume that we are sitting on a cell named @c cell1, whose neighbor behind
+   * the @c ith face is one level coarser. Let us name this coarser neighbor
+   * @c cell2. Then, by calling
+   * periodic_neighbor_of_coarser_periodic_neighbor(), from @c cell1, we get
+   * a @c face_num and a @c subface_num. Now, if we call
+   * periodic_neighbor_child_on_subface() from cell2, with the above face_num
+   * and subface_num, we get an iterator to @c cell1.
+   */
+  TriaIterator<CellAccessor<dim, spacedim> >
+  periodic_neighbor_child_on_subface (const unsigned int face_no,
+                                      const unsigned int subface_no) const;
+
+  /**
+   * This function is a generalization of
+   * periodic_neighbor_of_periodic_neighbor()
+   * for those cells which have a coarser periodic neighbor. The returned
+   * pair of numbers can be used in periodic_neighbor_child_on_subface()
+   * to get back to the current cell. In other words, the following
+   * assertion should be true, for a cell with coarser periodic neighbor:
+   * cell->periodic_neighbor(i)->periodic_neighbor_child_on_subface(face_no, subface_no)==cell
+   */
+  std::pair<unsigned int, unsigned int>
+  periodic_neighbor_of_coarser_periodic_neighbor (const unsigned face_no) const;
+
+  /**
+   * This function returns the index of the periodic neighbor. If there is
+   * no periodic neighbor at the given face, the returned value is -1.
+   */
+  int
+  periodic_neighbor_index (const unsigned int i) const;
```

Comment:
Yes, the index means the same thing as `cell->index()`. Since, this word is used in the documentation of the `neighbor_index()` function, I assume the users should understand what we mean by that. I fixed the documentation in the following way, though:
```
This function returns the index of the periodic neighbor at the
@c ith face of the current cell. If there is no periodic neighbor
at the given face, the returned value is -1.
```